src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/TagHelperResolver.cs

```diff
@@ -28,7 +28,7 @@ protected virtual async Task<TagHelperResolutionResult> GetTagHelpersAsync(Proje
                 throw new ArgumentNullException(nameof(engine));
             }
 
-            var providers = engine.Engine.Features.OfType<ITagHelperDescriptorProvider>().ToArray();
+            var providers = engine.Engine.Features.OfType<ITagHelperDescriptorProvider>().OrderBy(f => f.Order).ToArray();
```

Comment:
@rynowak, this bug was causing bind to run before component descriptor provider in design time.